graders/c/cgrader/cgrader.py

```diff
@@ -123,18 +132,18 @@ def run_command(self, command, input=None, sandboxed=True, timeout=None, env=Non
 
     def compile_file(
         self,
-        c_file,
-        exec_file=None,
-        add_c_file=None,
-        compiler=None,
-        flags=None,
-        pkg_config_flags=None,
+        c_file: list[str] | str,
+        exec_file: str | None = None,
+        add_c_file: str | list[str] | None = None,
+        compiler: str | None = None,
+        flags: str | list[str] | None = None,
+        pkg_config_flags: str | list[str] | None = None,
         add_warning_result_msg=True,
         ungradable_if_failed=True,
         return_objects=False,
         enable_asan=False,
```

Comment:
Did pyright complain about annotating this as a boolean?

Reply:
It did not -- I think it is inferred from the default parameters (assuming you are referring to `add_warning_result_msg` and related)

Reply:
Ahh I see, I'm in favor of adding as many explicit annotations as possible. It makes it easier to tell what's happening for someone reading the code, and the quality of the type inference is tool-dependent and generally pretty inconsistent.

Reply:
Resolved in [a339f22](https://github.com/PrairieLearn/PrairieLearn/pull/11119/commits/a339f2215dd488a1b52d4c17678b35d5ee8b4469)